core/src/main/java/com/yahoo/ycsb/workloads/GraphWorkload.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2018 YCSB contributors. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.workloads;
+
+import com.yahoo.ycsb.*;
+import com.yahoo.ycsb.generator.Generator;
+import com.yahoo.ycsb.generator.graph.Graph;
+import com.yahoo.ycsb.generator.graph.GraphComponent;
+import com.yahoo.ycsb.generator.graph.GraphDataGenerator;
+import com.yahoo.ycsb.generator.graph.Node;
+import com.yahoo.ycsb.generator.graph.randomcomponents.RandomGraphComponentGenerator;
+import com.yahoo.ycsb.generator.operationorder.OperationOrderGenerator;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+import static com.yahoo.ycsb.workloads.CoreWorkload.*;
+import static java.io.File.separatorChar;
+
+/**
+ * Workload class for graph data.
+ * <p>
+ * This workload will store the generated values in ycsbDir/benchmarkingData to be able to reproduce the
+ * benchmark run with the same data on multiple databases.
+ * You can change the directory with the parameter {@value DATA_SET_DIRECTORY_PROPERTY}.
+ * <p>
+ * Every node will have a size of 500 Bytes by default.
+ * This can be changed via the {@value NODE_BYTE_SIZE_PROPERTY} parameter.
+ * <p>
+ * The recordCount property determines how many nodes will be inserted. The total amount of database inserts could
+ * be higher due to edges being inserted.
+ */
+public class GraphWorkload extends Workload {
```

Comment:
please place graph workload stuff in its own module rather than mixing into the core workload module.